src/components/MyData/styles.ts

```diff
@@ -0,0 +1,59 @@
+import { StyleSheet } from 'react-native';
+import { VictoryThemeDefinition } from 'victory-core';
+import { createStyles } from '../BrandConfigProvider';
+
+type StyleExtractor<T> = T extends { style: unknown } ? T['style'] : never;
+type VictoryStyleProp<K extends keyof VictoryThemeDefinition> = StyleExtractor<
+  Required<Required<VictoryThemeDefinition>[K]>
+>;
+
+export const defaultChartStyles = createStyles('ChartStyles', (theme) => ({
+  independentAxis: {
+    grid: {
+      strokeDasharray: '0 2 0',
+      stroke: theme.colors.outline,
+    },
+    axis: {
+      stroke: theme.colors.onBackground,
+    },
+    tickLabels: {
+      padding: 8,
+      fill: theme.colors.onBackground,
+    },
+  } as VictoryStyleProp<'axis'>,
```

Comment:
I assume wrapping the T[P] with Partial was the trick to making this work correctly? Love it.

Reply:
Yeah exactly, without it, a user would have to define all fields of `tickLabels` if they were defining a `tickLabel` style.